pertpy/tools/_metrics_3g.py

```diff
@@ -0,0 +1,200 @@
+from collections.abc import Mapping
+from functools import partial
+from types import MappingProxyType
+from typing import TYPE_CHECKING, Any, Literal, Optional
+
+import anndata as ad
+import numpy as np
+import pandas as pd
+import pynndescent
+import scanpy as sc
+from scipy.sparse import issparse
+from scipy.sparse import vstack as sp_vstack
+from sklearn.base import ClassifierMixin
+from sklearn.linear_model import LogisticRegression
+
+from pertpy.tools._distances._distances import Distance
+
+if TYPE_CHECKING:
+    from numpy.typing import NDArray
+
+
+def compare_de(X: np.ndarray, Y: np.ndarray, C: np.ndarray, shared_top: int = 100, **kwargs) -> dict:
```

Comment:
Don't think we need that one, but if you can convince me, I think that we should move this to the DifferentialGeneExpression class